dsl_parser/tests/test_import_resolver.py

```diff
@@ -0,0 +1,206 @@
+                                     err_msg_regex=None):
+        if expected_resolver:
+            resolver = utils.create_import_resolver(resolver_configuration)
+            self.assertEqual(resolver.__class__, expected_resolver.__class__)
+            if expected_params_name:
+                self.assertEqual(
+                    resolver.__getattribute__(expected_params_name),
+                    expected_resolver.__getattribute__(expected_params_name))
+        else:
+            self.assertRaisesRegexp(utils.ResolverInstantiationError,
+                                    err_msg_regex,
+                                    utils.create_import_resolver,
+                                    resolver_configuration)
+
+    def test_no_configuration_specified(self):
+        self._test_create_import_resolver(
+            expected_resolver=DefaultImportResolver(),
+            expected_params_name=DEFAULT_RESLOVER_RULES_KEY)
+
+    def test_specified_default_class_path_and_params(self):
+        parameters = {
+            DEFAULT_RESLOVER_RULES_KEY: [{'rules1key': 'rules1value'}]
+        }
+        resolver_configuration = {
+            RESOLVER_IMPLEMENTATION_KEY: default_resolver_class_path,
+            RESLOVER_PARAMETERS_KEY: parameters
+        }
+        self._test_create_import_resolver(
+            resolver_configuration=resolver_configuration,
+            expected_resolver=DefaultImportResolver(**parameters),
+            expected_params_name=DEFAULT_RESLOVER_RULES_KEY)
+
+    def test_specified_default_class_path_no_params(self):
+        resolver_configuration = {
+            RESOLVER_IMPLEMENTATION_KEY: default_resolver_class_path
+        }
+        self._test_create_import_resolver(
+            resolver_configuration=resolver_configuration,
+            expected_resolver=DefaultImportResolver(),
+            expected_params_name=DEFAULT_RESLOVER_RULES_KEY)
+
+    def test_specified_params_no_class_path(self):
+        parameters = {
+            DEFAULT_RESLOVER_RULES_KEY: [{'rules1key': 'rules1value'}]
+        }
+        resolver_configuration = {
+            RESLOVER_PARAMETERS_KEY: parameters
+        }
+        self._test_create_import_resolver(
+            resolver_configuration=resolver_configuration,
+            expected_resolver=DefaultImportResolver(**parameters),
+            expected_params_name=DEFAULT_RESLOVER_RULES_KEY)
+
+    def test_create_custom_resolver(self):
+        parameters = {
+            'custom_reolver_parameters': {}
+        }
+        resolver_configuration = {
+            RESOLVER_IMPLEMENTATION_KEY: custom_resolver_class_path,
+            RESLOVER_PARAMETERS_KEY: parameters
+        }
+        self._test_create_import_resolver(
+            resolver_configuration=resolver_configuration,
+            expected_resolver=CustomImportResolver(
+                custom_reolver_parameters={}),
+            expected_params_name='parameters')
```

Comment:
is the name always expected to be parameters? if so, we can remove the expected_param_name entirely. If not, let's test another name